src/java.base/share/classes/sun/security/provider/HSS.java

```diff
@@ -552,6 +569,7 @@ byte[] lmotsPubKeyCandidate(LMSignature lmSig, byte[] message, LMSPublicKey pKey
                 preCandidate[21] = (byte) 0x80;
 
                 byte[] preZi = hashBuf.clone();
```

Comment:
hashBuf is assigned at the initialisation of the LMOTSParams object. If (when) we introduce more algorithms, the initialisation code and the digestFixedLengthPreprocessed() code needs to be changed only (its first parameter should be such that it can use the hash algorithm that the object would be initialised to use).